### PR TITLE
feat: api: add SectorNumber to MarketDealState (nv22)

### PR DESCRIPTION
**Pull request by rvagg**

### CODE REVIEW

**Comment by rvagg** [comment]
haha, same PR! and I even reviewed that one .. but I do see a bug in that other one

**Review by rjan90 — APPROVED** [review]

File: api/api_full.go [hunk 1]
```diff
@@ -1166,13 +1166,15 @@ type MarketBalance struct {
 }
 
 type MarketDealState struct {
-	SectorStartEpoch abi.ChainEpoch // -1 if not yet included in proven sector
-	LastUpdatedEpoch abi.ChainEpoch // -1 if deal state never updated
-	SlashEpoch       abi.ChainEpoch // -1 if deal never slashed
+	SectorNumber     abi.SectorNumber // 0 if not yet included in proven sector (0 is also a valid sector number).
```
Comment (LexLuthr) [inline api/api_full.go:1169]: Can we do this to -1. Having 0 as valid sector creates a lot of issues in checks. Boost already have a form on this issue for indexing and deal announcement where deal fails in AP(). I would love to have uniformity of -1 for sector number as much as possible.

Comment (rvagg) [reply api/api_full.go:1169]: sadly not if we're going to stick with `abi.SectorNumber`, and this was a limitation of FIP0076 too

```go
type SectorNumber uint64
```

It's in state as `0` so even if we introduced a new `int64` for this, we'd still be loading >=`0` from state to fetch this.

Comment (LexLuthr) [reply api/api_full.go:1169]: 😞 I can work with this. I will handle the exception on our side.

**Pull request merged**